### PR TITLE
Add carbon_defi_sei_base_trades

### PR DESCRIPTION
**Pull request by Hosuke**
### Update!
Please build spells in the proper [subproject](../dbt_subprojects/) directory. For more information, please see the main [readme](../README.md), which also links to a GH discussion with the option to ask questions.

### Contribution type
Please check the type of contribution this pull request is for:

- [ ] New spell(s)
- [x] Adding to existing spell lineage
- [ ] Bug fix

**Note:** You can safely discard any section below which doesn't apply based on selection above

---

### For new spell(s)
If you are building new spell(s), please provide the following information:

- **Spell name(s):** *schema.table_name*
- **Description:** [Detailed description of the new spell(s) and their purpose]
- **Who are the new spell(s) for?** [Internal team or general community]
- **How will the new spell(s) be used downstream?** [Description of downstream usage]
- **Implementation details:** [Information on how the spell(s) are implemented]
- **Test instructions:** [How to test the new spell(s)]
- **Related issue(s):** [Link to related issues, if any]

---

### For adding to existing spell lineage
If you are adding to an existing spell lineage, please provide the following information:

- **Description:** [Description of the changes made]

---

### For bug fixes
If you are fixing a bug, please provide the following information:

- **Description:** [Description of the bug fix]
- **Steps to reproduce:** [How to reproduce the bug]
- **Implementation details:** [Information on how the bug was fixed]
- **Test instructions:** [How to test the fix]
- **Related issue(s):** [Link to related issues, if any]

---

### Additional information
Please provide any additional information that might help us review your pull request:

- [Any additional information]

---

Thank you for your contribution!

### CODE REVIEW

File: dbt_subprojects/dex/models/trades/sei/platforms/carbon_defi_sei_base_trades.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        alias = 'base_trades',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['tx_hash', 'evt_index'],
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+{% set weth_address = '0x160345fc359604fc6e70e3c5facbde5f7a9342d8' %}
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/trades/sei/platforms/carbon_defi_sei_base_trades.sql:13]: I am not sure if this is the correct `WETH` address on sei:
https://seistream.app/contracts/0x160345fc359604fc6e70e3c5facbde5f7a9342d8

Comment (jeff-dude) [reply dbt_subprojects/dex/models/trades/sei/platforms/carbon_defi_sei_base_trades.sql:13]: yes, this is `weth`, but i think we actually want to use `WSEI` here as `SEI` is the native token to the chain, correct?

i think `weth` on `sei` is simply another erc20 that can be bridged across, but anything using `0xeee..` address would translate to `WSEI` for join to `prices.usd`

if we run both of these:
```
select targetToken, count(1)
from carbon_defi_sei.CarbonController_evt_TokensTraded
where targetToken = 0xE30feDd158A2e3b13e9badaeABaFc5516e95e8C7
group by targetToken
order by 2 desc

select targetToken, count(1)
from carbon_defi_ethereum.CarbonController_evt_TokensTraded
where targetToken = 0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2
group by targetToken
order by 2 desc
```

we see v little usage on the wrapped native token per chain, but if you remove the where clause, you can see the opposite side. on ethereum, lots of stablecoins and `0xeee..`. on sei, lots of stablecoins and `0xeee..`

Comment (jeff-dude) [reply dbt_subprojects/dex/models/trades/sei/platforms/carbon_defi_sei_base_trades.sql:13]: can we turn `weth_address` variable to a more generic `wrapped_native_token` variable?
then update sei to use `WSEI` [address](https://seitrace.com/token/0xE30feDd158A2e3b13e9badaeABaFc5516e95e8C7?chain=pacific-1)?
then we should double check our prices spells, ensure on sei chain, we have `WSEI` and `WETH` in tokens file

File: dbt_subprojects/dex/models/trades/sei/platforms/carbon_defi_sei_base_trades.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}}
+
+{% set wsei_address = '0xE30feDd158A2e3b13e9badaeABaFc5516e95e8C7' %}
+
+{{
+    carbon_defi_compatible_trades(
+        blockchain = 'sei',
+        project = 'carbon_defi',
+        CarbonController_evt_TokensTraded = source('carbon_defi_sei', 'CarbonController_evt_TokensTraded'),
+        wrapped_native_token = weth_address
```
Comment (Hosuke) [inline dbt_subprojects/dex/models/trades/sei/platforms/carbon_defi_sei_base_trades.sql:20]:
```suggestion
        wrapped_native_token = wsei_address
```

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**